File: 2027-minimum-moves-to-convert-string/2027-minimum-moves-to-convert-string.py
class Solution:
    def minimumMoves(self, s: str) -> int:
        ind, moves = 0, 0
        n = len(s)
        while ind < n:
            if s[ind] == 'X':
                ind += 3
                moves += 1
            else:
                ind += 1
        return moves

File: 2027-minimum-moves-to-convert-string/2027-minimum-moves-to-convert-string.cpp
class Solution {
public:
    int minimumMoves(string s) {
        //form the groups where the group starts from 'X'
        int cnt = 0;
        int i;
        for(i = 0; i < s.size()-2; i++) {
            if(s[i] == 'X') {
                //form the group, record
                cnt++;
                s[i+1] = s[i+2] = 'O';
            }
        }
        //here i becomes s.size()-3
        if(s[i] == 'X' || s[i+1] == 'X' || s[i+2] == 'X' ) cnt++;
        
        return cnt;
    }
};